platform/linux-generic/include/odp_packet_internal.h

```diff
@@ -248,6 +251,106 @@ static inline void packet_init(odp_packet_hdr_t *pkt_hdr, uint32_t len)
 	pkt_hdr->input = ODP_PKTIO_INVALID;
 }
 
+/**
+ * Check if copying packet metadata between pools is possible
+ *
+ * @param      src_pool     Source packet pool handle
+ * @param      dst_pool     Destination packet pool handle
+ *
+ * @retval  0 when possible without user area copy
+ * @retval >0 when possible with user area copy
+ * @retval <0 when not possible
+ */
+static inline int _odp_packet_copy_md_possible(odp_pool_t src_pool,
+					       odp_pool_t dst_pool)
+{
+	const pool_t *src_hdr;
+	const pool_t *dst_hdr;
+
+	if (src_pool == dst_pool)
+		return 0;
+
+	src_hdr = pool_entry_from_hdl(src_pool);
+	dst_hdr = pool_entry_from_hdl(dst_pool);
+
+	if (dst_hdr->param_uarea_size < src_hdr->param_uarea_size)
+		return -1;
+
+	return 1;
+}
+
+/**
+ * Copy packet metadata
+ *
+ * This function is assumed to never fail. Use _odp_packet_copy_md_possible() to
+ * check beforehand that copying packet metadata between source and destination
+ * packet pools is possible.
+ *
+ * @param      src_hdr     Source packet header
+ * @param      dst_hdr     Destination packet header
+ * @param      uarea_copy  Copy user area data. If false, user area pointers
+ *                         are swapped between the packet headers (allowed
+ *                         only when packets are from the same pool).
+ */
+static inline void _odp_packet_copy_md(odp_packet_hdr_t *src_hdr,
+				       odp_packet_hdr_t *dst_hdr,
+				       odp_bool_t uarea_copy)
```

Comment:
Usually, copy functions have "dst" on left, and "src" on right (analog to dst = src):
  void * memcpy ( void * destination, const void * source, size_t num );

_odp_packet_copy_md_possible() would need the same change.

Reply:
Fixed in V6.